src/kube.rs

```diff
@@ -184,72 +190,114 @@ fn run_individual(
+
+            let pod_info = PodInfo {
+                name: pod_name.clone(),
+                container: container,
+                parent: pod_name.clone(),
+                file_name: file_name,
+            };
+            pod_info_vec.push(pod_info);
+        }
     }
-    kube_cmd.arg("get");
-    kube_cmd.arg("pods");
-    kube_cmd.arg("-n");
-    kube_cmd.arg(&namespace);
-    kube_cmd.arg("-o");
-    kube_cmd.arg("jsonpath={range .items[*]}{.metadata.name} {.spec['containers', 'initContainers'][*].name}\n{end}");
 
-    let output = kube_cmd.output().expect("Failed to get kubernetes pods");
+    Ok(pod_info_vec)
+}
 
-    if output.stderr.len() != 0 {
-        let byte_string = String::from_utf8_lossy(&output.stderr);
-        utils::generate_err(byte_string.to_string())?
+/// An informer that will update the main thread pool if a new pod is spun up.
+pub async fn pod_informer() -> Result<(), Box<dyn ::std::error::Error>> {
+    let client = get_kube_client().await;
+    let events = Api::v1Event(client);
+    let ei = Informer::new(events).init().await?;
+    loop {
+        let mut events = ei.poll().await.unwrap().boxed();
+
+        while let Some(event) = events.next().await {
+            let event = event?;
+            handle_events(event).await?;
+        }
     }
-
-    let pods = str::from_utf8(&output.stdout)?;
-    let pods_vec: Vec<&str> = pods.split("\n").collect();
-    Ok(pods_vec.iter().map(|&x| x.to_string()).collect())
 }
 
-/// We generate a hashmap with the key being the file_path (easy access to write files), and the
-/// value being a PodInfo struct
-fn generate_hashmap(pod_vec: Vec<String>, outfile: &str) -> HashMap<String, PodInfo> {
-    let mut pods_hashmap = HashMap::new();
-    for info in pod_vec {
-        if info == "" {
-            continue;
+/// Watches for an event.  If there is a new added event, we check if its a created pod type.  If
+/// it is we see if the pod exists in the clusters namespace, and if it does exist, we make sure
+/// the pod is healthy.  If the pod is healthy, we had it to the threadpool and begin tailing the
+/// containers in the pod
+async fn handle_events(ev: WatchEvent<v1Event>) -> Result<(), Box<dyn ::std::error::Error>> {
+    match ev {
+        WatchEvent::Added(o) => {
+            if o.message.contains("Created pod") {
+                println!("{}, checking to see if this event effects wufei", o.message);
+
+                let pod_message: Vec<&str> = o.message.split(":").collect();
+                let pod_str = pod_message[1].trim();
+
+                let pods = get_all_pod_info().await?;
+                for p in pods {
+                    if pod_str == p.name {
+                        loop {
+                            let healthy = check_status(&p.name).await?;
+                            if healthy {
+                                break;
+                            }
+                            let five_secs = time::Duration::from_secs(5);
```

Comment:
would you ever want to expose this to the user?

Reply:
I thought about it, and maybe even adding a verbose flag to allow that to happen.  Havent made up my mind yet.